infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/resolver/AbstractServerResolver.java

```diff
@@ -40,6 +43,29 @@ public AbstractServerResolver(Iterable<Service> services) {
     }
   }
 
```

Comment:
since it public can you explain what is going on here?

Reply:
test for this method?

Reply:
It's a copy of the code from IngressResolver. But you're right - I'll add the javadoc and some quick tests.